ext/search_kit/Civi/Api4/Action/SearchDisplay/GetSearchTasks.php

```diff
@@ -19,6 +19,18 @@ class GetSearchTasks extends \Civi\Api4\Generic\AbstractAction {
    */
   protected $entity;
 
+  /**
+   * Name of search
+   * @var string
+   */
+  protected $search;
```

Comment:
OK! Good start. However, this will be slightly more complicated because when running a display in preview mode it hasn't necessarily been saved yet, and won't have a name. This is where the `SavedSearchInspectorTrait` comes into play. So I think this class should:

1. Use that trait.
2. Remove this var (it's already in the trait)
3. Keep the below var
4. At the beginning of `run` it should call the `loadSavedSearch()` and `loadSearchDisplay()` functions.
5. That will transform `savedSearch` and `display` params into arrays, and the full arrays ought to be passed to the hook.
6. I see that you're only passing `savedSearch` to the hook. You should pass `display` as well.

Reply:
Thanks for the pointers. That works. I've updated the PR and now see two arrays. So you can choose to match on search or individual display. If not saved "name" is not defined in the array.

Reply:
@mattwire here's a followup to make sure `name` and `id` array keys are present even if not defined. #25146

Reply:
Ok I merged that